moped-toolbox/council_districts_load/insert_council_districts.py

```diff
@@ -0,0 +1,124 @@
+"""Downloads council districts geojson from AGOL and inserts it into Moped"""
+import argparse
+import requests
+from secrets import HASURA
+
+
+DISTRICTS_URL = "https://services.arcgis.com/0L95CJ0VTaxqcmED/ArcGIS/rest/services/BOUNDARIES_single_member_districts/FeatureServer/0/query?where=1%3D1&objectIds=&time=&geometry=&geometryType=esriGeometryEnvelope&inSR=&spatialRel=esriSpatialRelIntersects&resultType=none&distance=0.0&units=esriSRUnit_Meter&relationParam=&returnGeodetic=false&outFields=*&returnGeometry=true&returnCentroid=false&featureEncoding=esriDefault&multipatchOption=xyFootprint&maxAllowableOffset=&geometryPrecision=8&outSR=4326&defaultSR=&datumTransformation=&applyVCSProjection=false&returnIdsOnly=false&returnUniqueIdsOnly=false&returnCountOnly=false&returnExtentOnly=false&returnQueryGeometry=false&returnDistinctValues=false&cacheHint=false&orderByFields=&groupByFieldsForStatistics=&outStatistics=&having=&resultOffset=&resultRecordCount=&returnZ=false&returnM=false&returnExceededLimitFeatures=true&quantizationParameters=&sqlFormat=none&f=pgeojson&token="
+
+MUTATION = """
+mutation InsertCouncilDistricts($objects: [layer_council_district_insert_input!]!) {
+  insert_layer_council_district(objects: $objects) {
+    returning {
+      id
+    }
+  }
+}
+"""
+
+
+def make_hasura_request(*, query, env, variables=None):
+    """Fetch data from hasura
+
+    Args:
+        query (str): the hasura query
+        variables (dict): optional variables to be included
+        env (str): the environment name, which will be used to acces secrets
+
+    Raises:
+        ValueError: If no data is returned
+
+    Returns:
+        dict: Hasura JSON response data
+    """
+    admin_secret = HASURA["hasura_graphql_admin_secret"][env]
+    endpoint = HASURA["hasura_graphql_endpoint"][env]
+    headers = {
+        "X-Hasura-Admin-Secret": admin_secret,
+        "content-type": "application/json",
+    }
+    payload = {"query": query, "variables": variables}
+    res = requests.post(endpoint, json=payload, headers=headers)
+    res.raise_for_status()
+    data = res.json()
+    try:
+        return data["data"]
+    except KeyError:
+        raise ValueError(data)
+
+
+def get_council_districts_geojson():
+    res = requests.get(DISTRICTS_URL)
+    res.raise_for_status()
+    return res.json()["features"]
+
+
+def make_multipoly(features):
+    for feature in features:
+        if feature["geometry"]["type"] == "Polygon":
+            feature["geometry"]["type"] = "MultiPolygon"
+            feature["geometry"]["coordinates"] = [feature["geometry"]["coordinates"]]
+            print(
+                f"Converted district {feature['properties']['COUNCIL_DISTRICT']} to MuliPolygon"
+            )
```

Comment:
the Esri feature type for this layer is `esriGeometryPolygon`, which is agnostic to multi vs single-part shapes. when the AGOL server converts the features to geojson it returns either `MultiPolygon` or `Polygon` as needed.

we're using the `MultiPolygon` type in the DB, so so the hack to convert poly to a multipoly is to simply wrap the poly's coordinates in `[]`.